oauth2-server-snap/lib/Network/OAuth2/Server/Snap.hs

```diff
@@ -44,7 +49,16 @@ authorizeEndpoint = writeText "U AM U?"
 tokenEndpoint
     :: Handler b (OAuth2 IO b) ()
 tokenEndpoint = do
-    let grant_type = GrantPassword
+    clientAuth
+    req <- getRequest
+    let grant_type'' = S.lookup "grant_type" (headers req)
```

Comment:
Isn't `grant_type` a form encoded variable in the request body? And not a header.